### PR TITLE
  Adding examples for FTS query timeouts

### PR DESCRIPTION
**Pull request by sreekanth-cb**

### CODE REVIEW

**Comment by sreekanth-cb** [comment]
Tagging @amarantha-k , @abhinavdangeti

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 1]
```diff
@@ -100,6 +100,37 @@ http://localhost:18094/api/index/travel-sample-index/query \
 For additional information on secure communication with Couchbase Server, see xref:manage:manage-security/manage-connections-and-disks.adoc[Manage Connections and Disks].
 
 
+[#searching-with-query-timeouts]
+== Searching with timeouts
+
+Once the the index becomes really large, it is possible that certain search requests may take
+longer time to finish the execution.Such slow responses would affect the query latencies as well as the resource consumption on the server side.
```
Comment (abhinavdangeti) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: Space after period.

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 2]
```diff
@@ -100,6 +100,37 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+
+For avoiding such unplesant situation, users can pass a search "timeout" value in the request.
```
Comment (abhinavdangeti) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: Typo: unpleasant* situations*

Feels a bit harsh though, perhaps we should drop the entire line..

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 3]
```diff
@@ -100,6 +100,37 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+By specificing timeout parameters in the seach request, users are bound to get a response within the stipulated time.
+Here the timeout value is specified in unit of milliseconds.
```
Comment (abhinavdangeti) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: Refactor: "..the timeout specified is in milliseconds."

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 4]
```diff
@@ -100,6 +100,37 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+
+If the search request times out before gathering all results, then the partial results will be
+returned to the user with detailed partition level search result status.
```
Comment (abhinavdangeti) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc:114]: What could be useful here is an output snippet indicating the partial search status too? We can achieve this by placing sleeps in the code.

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 5]
```diff
@@ -100,6 +100,37 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+In case of timeouts, the response's http status code would be 412.
```
Comment (abhinavdangeti) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: (412: Precondition Failed)

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc (hunk 4 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-searching-with-the-rest-api.adoc:114]: Thought of that, but then realised that once the error comes, it's natural for the user to interpret it like any other response.  Response always follows the same pattern and describing that in its respective section ought to be enough is my thinking.
The intent here was more on - there is an option to do this and this is how you do it..

**Comment by sreekanth-cb** [comment]
thanks for the comments @abhinavdangeti , Done!

**Comment by abhinavdangeti** [comment]
Looks good @sreekanth-cb . 
There seems to be a "Merge" commit that seems to have snuck in?

**Comment by sreekanth-cb** [comment]
Will recreate it as my squash commit attempts failed..

**Pull request closed without merging**